sbot/simulator/camera.py

```diff
@@ -0,0 +1,73 @@
+import struct
+
+import cv2
+import numpy as np
+from april_vision import FrameSource
+from numpy.typing import NDArray
+from serial import serial_for_url
+
+from ..utils import BoardInfo
+
+
+class WebotsRemoteCameraSource(FrameSource):
+    # Webots cameras include an alpha channel, this informs april_vision of how to handle it
+    COLOURSPACE = cv2.COLOR_BGRA2GRAY
+
+    def __init__(self, camera_info: BoardInfo) -> None:
+        self.calibration = (0.0, 0.0, 0.0, 0.0)
+        # Use pyserial to give a nicer interface for connecting to the camera socket
+        self._serial = serial_for_url(camera_info.url, baudrate=115200, timeout=None)
+
+        # Check the camera is connected
+        response = self._make_request("*IDN?")
+        if not response.split(b":")[1].lower().startswith(b"cam"):
+            raise RuntimeError(f"Camera not connected to a camera, returned: {response!r}")
+
+        # Get the calibration data for this camera
+        response = self._make_request("CAM:CALIBRATION?")
+
+        # The calibration data is returned as a string of floats separated by colons
+        new_calibration = tuple(map(float, response.split(b":")))
+        assert len(new_calibration) == 4, f"Invalid calibration data: {new_calibration}"
+        self.calibration = new_calibration
+        assert len(self.calibration) == 4, f"Invalid calibration data: {self.calibration}"
```

Comment:
Nit: This second assertion isn't necessarily, as it will be confirming what was already confirmed 2 lines above.
```suggestion
```